fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateDisassemble.java

```diff
@@ -64,49 +65,69 @@ public Rule<Plan> build() {
+                    }
+                });
+            }
+
+            List<Expression> updateGroupByExpressionList = groupByExpressionList;
+            List<NamedExpression> updateGroupByAliasList = updateGroupByExpressionList.stream()
+                    .map(g -> new Alias<>(g, g.sql()))
+                    .collect(Collectors.toList());
+
+            List<NamedExpression> updateOutputExpressionList = Lists.newArrayList();
+            updateOutputExpressionList.addAll(updateGroupByAliasList);
+            updateOutputExpressionList.addAll(aggregateFunctionAliasMap.values());
+
+            List<Expression> mergeGroupByExpressionList = updateGroupByAliasList.stream()
+                    .map(NamedExpression::toSlot).collect(Collectors.toList());
+
+            List<NamedExpression> mergeOutputExpressionList = Lists.newArrayList();
+            for (NamedExpression o : outputExpressionList) {
+                if (o.contains(AggregateFunction.class::isInstance)) {
+                    mergeOutputExpressionList.add((NamedExpression) new AggregateFunctionParamsRewriter()
+                            .visit(o, aggregateFunctionAliasMap));
+                } else {
+                    for (int i = 0; i < updateGroupByAliasList.size(); i++) {
+                        // TODO: we need to do sub tree match and replace. but we do not have semanticEquals now.
+                        //    e.g. a + 1 + 2 in output expression should be replaced by
+                        //    (slot reference to update phase out (a + 1)) + 2, if we do group by a + 1
+                        //   currently, we could only handle output expression same with group by expression
+                        if (o instanceof SlotReference) {
+                            // a in output expression will be SLotReference
+                            if (o.equals(updateGroupByExpressionList.get(i))) {
+                                mergeOutputExpressionList.add(updateGroupByAliasList.get(i).toSlot());
+                                break;
+                            }
+                        } else if (o instanceof Alias) {
+                            // a + 1 in output expression will be Alias
+                            if (o.child(0).equals(updateGroupByExpressionList.get(i))) {
+                                mergeOutputExpressionList.add(updateGroupByAliasList.get(i).toSlot());
+                                break;
+                            }
+                        }
                     }
                 }
-                intermediateAggExpressionList.add(namedExpression);
             }
+
             LogicalAggregate localAgg = new LogicalAggregate(
-                    agg.getGroupByExprList().stream().map(Expression::clone).collect(Collectors.toList()),
-                    intermediateAggExpressionList,
+                    updateGroupByExpressionList,
+                    updateOutputExpressionList,
```

Comment:
I think this variable names and compute logic is confuse, how about this:
1. localGroupByExprs = originGloupByExprs
2. localOutputExprs = originOutput.withAlias
3. globalGroupByWithAlias = originGloupByExprs.withAlias
4. globalOutputAlias = originOutput.replaceAggregateFunctionArguments

this advantage is
1. variableName contains position and member information
2. assign statement contains the most simple compute logical

Reply:
i use the names in the stale planner. imo, local and global is better than update and merge. but as discussion before, we want to reuse names in stale planner as much as possible. so i reserve these names.

Reply:
okey, I remember these term. Keep this logic is reasonable, If reserve these names:

updateGroupByExprs = originGloupByExprs
updateOutputExprs = originOutput.withAlias
mergeGroupByWithAlias = originGloupByExprs.withAlias
mergeOutputWithAlias = originOutput.replaceAggregateFunctionArgumentsToAliasReference